### PR TITLE
GEOMESA-2157 Replacing joda with java time

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Deleting unused geomesa-utils TimeSnap, MinMaxTimeVisitor and csv package

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: docs/user/upgrade.rst [hunk 1]
```diff
@@ -34,6 +34,20 @@ been standardized . New parameters are outlined in the individual data store pag
...
+
+With the introduction of ``java.time`` in Java 8, the Joda Time project has been deprecated. As such, GeoMesa
+has removed its Joda dependency in favor of ``java.time``. One consequence of this is that custom date patterns
+in ``geomesa-convert`` are interpreted slightly differently. See `DateTimeFormatter`__ for details.
+
+__ https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html
+
+.. warning::
+
+  In particular, "year of era" has changed from ``Y`` to ``y``. ``Y`` now means "week-based year", and will
```
Comment (jahhulbert-ccri) [inline docs/user/upgrade.rst]: ouch

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/legacy/attribute/AttributeWritableIndex.scala [hunk 2]
```diff
@@ -105,7 +108,15 @@ object AttributeWritableIndex extends LazyLogging {
...
-  private val dateFormat     = ISODateTimeFormat.dateTime()
+
+  private val dateFormat =
+    new DateTimeFormatterBuilder()
+        .append(DateTimeFormatter.ISO_LOCAL_DATE_TIME)
+        .appendFraction(ChronoField.MILLI_OF_SECOND, 3, 3, true)
+        .optionalStart()
+        .appendOffsetId()
+        .toFormatter(Locale.US)
+        .withZone(ZoneOffset.UTC)
```
Comment (jahhulbert-ccri) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/legacy/attribute/AttributeWritableIndex.scala]: i can't believe they didn't add this as a default format

File: pom.xml [hunk 3]
```diff
@@ -350,11 +350,6 @@
                 <artifactId>joda-time</artifactId>
                 <version>2.3</version>
             </dependency>
```
Comment (jahhulbert-ccri) [inline pom.xml]: joda still made it in :)

Comment (jahhulbert-ccri) [reply pom.xml]: is that because we just want to manage it?

Comment (elahrvivaz) [reply pom.xml]: yeah, i'll add a comment - version 2.3 is pulled in by scalatra (we should probably upgrade scalatra too), and 2.7 is pulled in by osm4j. didn't want to file a new cq, so...

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/index/XZConfigurationTest.scala [hunk 4]
```diff
@@ -24,7 +24,7 @@ class XZConfigurationTest extends Specification with TestWithDataStore {
 
   val features = (0 until 10).map { i =>
     val sf = new ScalaSimpleFeature(sft, s"$i")
-    sf.setAttributes(Array[AnyRef](s"name$i", s"2010-05-07T$i:00:00.000Z",
+    sf.setAttributes(Array[AnyRef](s"name$i", s"2010-05-07T0$i:00:00.000Z",
```
[the remark this answers is not in the document]
Comment (elahrvivaz) [reply geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/index/XZConfigurationTest.scala]: wilco

[the remark this answers is not in the document]
Comment (elahrvivaz) [reply geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/index/XZConfigurationTest.scala]: now we're good as long as we stay under 24 ;)

File: pom.xml (hunk 3 above)
Comment (jahhulbert-ccri) [reply pom.xml]: sounds good. Yeah I think that some scalatra upgrades could be good for 1.4.0 maybe actually a version review of a few things...I was specifically looking at 2.11.8 bc of the repl tab completion improvements but that's more stuff I guess...also tracks the spark version.

**Comment by jahhulbert-ccri** [comment]
This looks pretty good to me...building now. I guess Java 8 time isn't QUITE as nice as joda for the record but it does seem that Joda stopped doing releases in favor of Java8 time. Seems ready for merging.

**Pull request merged**